android/src/main/java/com/swmansion/rnscreens/ScreenFragment.java

```diff
@@ -66,6 +64,63 @@ public Screen getScreen() {
     return mScreenView;
   }
 
+  public void onContainerUpdate() {
+    ScreenStackHeaderConfig config = findHeaderConfig();
+    boolean configInChild = hasChildScreenWithConfig(getScreen());
+    if (!configInChild && config != null && config.getScreenFragment().getActivity() != null) {
+      // if there is no child with config, we look for a parent with config to set the orientation
```

Comment:
I'm still a bit confused by this. Why we check if children have config while `findHeaderConfig` searches for the config in current container or in its parents? If it isn't trivial to explain maybe we can discuss over call and figure out best way to describe it here.

Reply:
Let's have a structure of `ScreenStack` -> `ScreenContainer` -> `ScreenStack`. We do this check in case the update of the container was triggered after setting the orientation in the child stack, which would result in setting the orientation of the parent `ScreenStack` even though the child `ScreenStack` should have priority. At the same time, we want to have an option to change the orientation on every update of container, because the middle `ScreenContainer` can have children that are not `ScreenStack`s, so after navigating from a child that is a `ScreenStack` to a child that is not, we want the orientation to be taken and applied from the parent `ScreenStack`.

Is this explanation clear enough?

Reply:
Also, it is probably good to change the code to only look for the parent `Screen` with `config` if there is no child with `config` to avoid traversing the hierarchy when not needed.